docs/developer/website-development.mdx

```diff
@@ -12,3 +20,320 @@ For more elaborate changes, including adding new pages, you will need to clone t
+
+#### ShowIfs/ShowIf
+`ShowIfs` and `ShowIf` form a `Tab` and a number of `TabItem`. They are defined in `src/components/LinguaFrancaMultiTargetUtils/ShowIf.tsx`.
+
+Specifically - you MUST use `ShowIf` inside of a `ShowIfs`, and a `ShowIfs` MUST contain at least one `ShowIf` as its child.
+
+##### Usage
+```jsx
+    <ShowIfs>
+        <ShowIf c>
+            {/* Markdown contents */}
+        </ShowIf>
+        <ShowIf cpp py ts>
+            {/* Markdown contents */}
+        </ShowIf>
+    </ShowIfs>
+```
+Where the target languages could be a subset of all targets which we support but must NOT collide. 
+As long as you have more than one `ShowIf` inside you can have as many as you like.
+
+##### Example
+```jsx
+    <ShowIfs>
+        <ShowIf c>
+            This is shown when the C target is chosen.
+        </ShowIf>
+        <ShowIf cpp py ts>
+            C++, Python and Typescript-related content is shown here.
+        </ShowIf>
+    </ShowIfs>
+    {/* No Rust here! If you choose Rust from the language selector, you will get what you chose before you chose Rust. */}
+```
+
+:::note[Rendered as]
+
+<ShowIfs>
+    <ShowIf c>
+        This is shown when the C target is chosen.
+    </ShowIf>
+    <ShowIf cpp py ts>
+        C++, Python and Typescript-related content is shown here.
+    </ShowIf>
+</ShowIfs>
+
+:::
+
+#### ShowOnly
+
+:::warning
+`ShowOnly` uses hacky Docusaurus-internal methods for the time being. To ensure better compatibility, you should prefer `ShowIf`.
+:::
+
+Sometimes, you want to display a piece of information only for one language. While you can use `ShowIfs` and one `ShowIf` inside of it, the issue people typically discover is that the rendered result looks miserable, because each `Tab` will add some sort of padding above and below it. 
+
+For this, we built `ShowOnly` which reads the selected tab from `@docusaurus/theme-common/internal/useTabs` and displays the information inside if the target selected matches, and the output is a `<div>` or a `<span>` so that it looks like it's blended in. This is hacky and could break any time if docusaurus decides to change how their `Tabs` work.
+
+##### Usage
+```jsx
+    <ShowOnly c cpp
+    inline>
+    {/* Markdown contents */}
+    </ShowOnly>
+```
+Where the target languages could be a subset of all targets which we support.  
+If you include `inline` it will result in a `<span>`, if not, a `<div>`.
+
+##### Examples
+```jsx
+The language which you used yells:
+<ShowOnly c cpp >Segfault!!!!!!</ShowOnly>
+```
+
+:::note[Rendered as]
+
+The language which you used yells:
+<ShowOnly c cpp >Segfault!!!!!!</ShowOnly>
+
+:::
+
+```jsx
+I am <ShowOnly c cpp inline>not</ShowOnly> memory safe!
+```
+
+:::note[Rendered as]
+
+I am <ShowOnly c cpp inline>not</ShowOnly> memory safe!
+
+:::
+
+#### ShowIfsInline
+
+:::warning
+`ShowIfsInline` uses hacky Docusaurus-internal methods for the time being. To ensure better compatibility, you should prefer `ShowIf`.
+:::
+
+:::danger
```

Comment:
We can leave it, but this "danger" note shouldn't be shown on the website, so let's remove it.